File: VXG.MediaPlayer.Android/src/Eclipse/MediaPlayerSDKTest_Simple/src/veg/mediaplayer/sdk/test/simple/MainActivity.java
/*
 *
  * Copyright (c) 2011-2017 VXG Inc.
 *
 */

package veg.mediaplayer.sdk.test.simple;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;
import android.view.View.OnClickListener;
import android.view.View.OnTouchListener;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.view.inputmethod.InputMethodManager;
import android.content.SharedPreferences;
import android.view.*;
import android.widget.RelativeLayout;
import android.widget.TextView.OnEditorActionListener;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;

import java.nio.ByteBuffer;
import java.util.Set;
import java.util.HashSet;
import java.util.ArrayList;

import android.preference.PreferenceManager;
import veg.mediaplayer.sdk.MediaPlayer;
import veg.mediaplayer.sdk.MediaPlayer.PlayerNotifyCodes;
import veg.mediaplayer.sdk.MediaPlayer.PlayerState;
import veg.mediaplayer.sdk.MediaPlayerConfig;

public class MainActivity extends Activity implements MediaPlayer.MediaPlayerCallback
{
	private MediaPlayer player = null;
	
	@Override
    public void onCreate(Bundle savedInstanceState) 
	{
		setTitle(R.string.app_name);
		super.onCreate(savedInstanceState);

		setContentView(R.layout.main);
		
		player = (MediaPlayer)findViewById(R.id.playerView);
		
		final ImageView pauseIndicator = (ImageView)findViewById(R.id.pauseIndicator);

    	MediaPlayerConfig conf = new MediaPlayerConfig();
    	conf.setDecodingType(0);
    	conf.setConnectionUrl("rtsp://184.72.239.149/vod/mp4:BigBuckBunny_115k.mov");
    	//conf.setConnectionUrl("rtsp://10.20.16.36/1.ts");
       	conf.setStartOffest(60000);

    	player.Open(conf, this);
    	
        player.setOnTouchListener(new OnTouchListener() 
		{
			public boolean onTouch(View v, MotionEvent event) 
			{
				switch(event.getAction())
				{
					case MotionEvent.ACTION_DOWN:
					{
						if (player != null && PlayerState.forType(player.getState()) > PlayerState.forType(PlayerState.Opened) && 
								PlayerState.forType(player.getState()) < PlayerState.forType(PlayerState.Closing))
						{
							
							if (player.getState() == PlayerState.Started)
							{
								player.Pause();
								pauseIndicator.setVisibility(View.VISIBLE);
							}
							else
							{
				    	    	player.Play();
								pauseIndicator.setVisibility(View.GONE);
							}
						}
					}
				}
				
				return true;
			}
		});
    	
	}

  	@Override
  	protected void onDestroy() 
  	{
  		Log.e(TAG, "onDestroy()");
  		player.Close();

		super.onDestroy();
   	}	
	
	private Handler handler = new Handler() 
    {
	        @Override
	        public void handleMessage(Message msg) 
	        {
	        	PlayerNotifyCodes status = (PlayerNotifyCodes) msg.obj;
	        }
	};
	
	// callback from Native Player 
	public int Status(int arg)
	{
    	Log.e(TAG, "Form Native Player status: " + arg);
		Message msg = new Message();
		msg.obj = PlayerNotifyCodes.forValue(arg);

		if (handler != null)
			handler.sendMessage(msg);
		
    	return 0;
    }

	@Override
	public int OnReceiveData(ByteBuffer buffer, int size, long pts) 
	{
		// TODO Auto-generated method stub
		return 0;
	}

	private static final String TAG = "MediaPlayerSDKTest.Simple";
}

File: VXG.MediaPlayer.Android/src/Xamarin/PlayerTestXamarin.Android/PlayerTestXamarin/PlayerTestXamarin.Android/MainActivity.cs
using System;
using Android.App;
using Android.Graphics;
using Android.OS;
using Android.Util;
using Android.Widget;
using Java.Nio;
using Veg.Mediaplayer.Sdk;

namespace PlayerTestXamarin.Droid
{
    [Activity(Label = "PlayerTestXamarin.Android", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity, MediaPlayer.IMediaPlayerCallback
    {
        private const string Url = "rtsp://184.72.239.149/vod/mp4:BigBuckBunny_115k.mov";

        private MediaPlayer _mediaPlayer;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetContentView(Resource.Layout.Main);

            _mediaPlayer = FindViewById<MediaPlayer>(Resource.Id.playerView);
            _mediaPlayer.SurfaceView.SetZOrderOnTop(true);

            // Not sure if this is not redundant
            var trackHolder = _mediaPlayer.SurfaceView.Holder;
            trackHolder.SetFormat(Format.Transparent);

            var connectButton = FindViewById<Button>(Resource.Id.connectButton);
            connectButton.Click += ConnectButton_Click;
        }

        private void ConnectButton_Click(object sender, EventArgs e)
        {
            _mediaPlayer.Open(new MediaPlayerConfig
            {
                ConnectionUrl = Url,
                ConnectionNetworkProtocol = -1,
                ConnectionBufferingTime = 300,
                ConnectionDetectionTime = 1000,
                DecodingType = 1,
                RendererType = 1,
                SynchroEnable = 1,
                SynchroNeedDropVideoFrames = 1,
                EnableColorVideo = 1,
                EnableAspectRatio = 1,
                DataReceiveTimeout = 30000,
                NumberOfCPUCores = 0
            }, this);
        }

        protected override void OnPause()
        {
            base.OnPause();

            _mediaPlayer?.OnPause();
        }

        protected override void OnStart()
        {
            base.OnStart();

            _mediaPlayer?.OnStart();
        }

        protected override void OnStop()
        {
            base.OnStop();

            _mediaPlayer?.OnStop();
        }

        public override void OnBackPressed()
        {
            base.OnBackPressed();

            _mediaPlayer?.Close();
        }

        public override void OnWindowFocusChanged(bool hasFocus)
        {
            base.OnWindowFocusChanged(hasFocus);

            _mediaPlayer?.OnWindowFocusChanged(hasFocus);
        }

        public override void OnLowMemory()
        {
            base.OnLowMemory();

            _mediaPlayer?.OnLowMemory();
        }

        protected override void OnDestroy()
        {
            base.OnDestroy();

            _mediaPlayer?.OnDestroy();
        }

        public int OnReceiveData(ByteBuffer p0, int p1, long p2)
        {
            Log.Error("PlayerTestXamarin", "Form Native Player OnReceiveData: size: " + p1 + ", pts: " + p2);

            return 0;
        }

        public int Status(int p0)
        {
            Log.Error("PlayerTestXamarin", "From Native Player status: " + p0);

            return 0;
        }
    }
}
